### PR TITLE
Pdm cache fix

### PR DESCRIPTION
**Pull request by freider**
## Describe your changes
Fix issue where all pdm cached packages would be automounted (MOD-2485)

The underlying issue was that pdm with cache symlinks the actual package data into site-packages, and we called `resolve()` on the package paths before comparing them with SYS_PREFIXES.

This patch changes the logic to preserve the un-expanded symlink paths for purposes of automount evaluation, but still resolves paths for serve/watch and deduplication

## Changelog

* Fix issue with pdm where all installed packages would be automounted when using package cache (MOD-2485)

### CODE REVIEW

**Comment by freider** [comment]
@prbot approve

File: modal/mount.py [hunk 1]
```diff
@@ -120,10 +120,10 @@ class _MountDir(_MountEntry):
     recursive: bool
 
     def description(self):
-        return str(self.local_dir)
+        return str(self.local_dir.resolve().expanduser())
```
Comment (gimaker) [inline modal/mount.py:123]: this looks problematic?

```
In [22]: pathlib.Path("~/foo").resolve().expanduser()
Out[22]: PosixPath('/home/staffan/src/modal/~/foo')

In [23]: pathlib.Path("~/foo").expanduser().resolve()
Out[23]: PosixPath('/home/staffan/foo')
```

**Review by gimaker — COMMENTED** [review]
:heavy_check_mark: lgtm overall, but see comment about `expanduser()` and `resolve()` interacting weirdly

File: modal/mount.py (hunk 1 above)
Comment (mwaskom) [reply modal/mount.py:123]: +1

**Pull request merged**